griptape/tasks/base_task.py

```diff
@@ -181,7 +185,18 @@ def after_run(self) -> None:
             )
 
     def can_run(self) -> bool:
-        return self.state == BaseTask.State.PENDING and all(parent.is_finished() for parent in self.parents)
+        # If this Task has been skipped or is not pending, it should not run
+        if self.is_skipped() or not self.is_pending():
+            return False
+
+        # If this Task has parents, and _all_ of them are skipped, it should not run
+        if self.parents and all(parent.is_skipped() for parent in self.parents):
+            return False
```

Comment:
need to set `self.state = BaseTask.State.SKIPPED` here?

Reply:
Depends what we want the behavior to be. Should a Task of entirely skipped parents be `SKIPPED` or `PENDING`?

I can see the argument for `SKIPPED` but this feels like the wrong place to do it.

Reply:
yeah modifying state here feels odd. might need to traverse down the graph as part of the `BranchTask` then

Reply:
I'm leaning towards this being fine. It's so much easier to implement it here than elsewhere.

Reply:
still need to set it?

Reply:
Oops